### DISCUSSION TITLE
How to setup the development environment?

### DISCUSSION THREAD

**Discussion by 4Source** [st3v3nmw/obsidian-spaced-repetition#850] [Q&A] [answered]
If have tried to setup this plugin for developing. But I can't get it to work. Can someone explain what are the required steps. 
Started with creating a fork than pull this repository.

**Comment by ronzulu** [comment, marked as answer]
Hi @4Source 

This is the only GitHub project I've worked on, and I setup my environment maybe 6 months ago, so I'm not sure of the steps.

I think I:
Went to https://github.com/st3v3nmw/obsidian-spaced-repetition

Created a new fork:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/2fd42270-a7e7-422e-85a9-17721edb8469)

Then from my fork:
https://github.com/ronzulu/obsidian-spaced-repetition

GitHub had automatically created the master branch, but I always create a new branch when working on an issue or experimenting.

https://github.com/ronzulu/obsidian-spaced-repetition/branches
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/ca04db03-f342-46bf-b811-e9c56383d207)

From within a branch:
https://github.com/ronzulu/obsidian-spaced-repetition/tree/feat-770-osr-reading-view

(1) confirmation of the original repository (or "upstream")
(2) Shows current branch and enables switching
(3) Accessing the code

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/23c0d54f-383c-4c9c-b719-ea9dd9008f71)

There are different methods for accessing the code, I use the GitHub Desktop method:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/acddf5eb-5232-4283-9340-f6e466fa192f)

And I've been using Visual Studio Code as the development environment under Windows.

Hope that helps

Cheers
Ronny

**Reply by 4Source** [reply]
Thanks for your answer. I think I didn't explain clearly what my problem is. The GitHub side is not my problem I forked the repo, created a new branch and pulled it to my computer. Here is where my problem starts. 
I have seen not npm used instead pnpm. I've never used this before, but at first glance it seems pretty much the same.
How should I set it up on my computer to create changes and test them? The main.js get created in the build folder where obsidian can't find it. Do I need to move this file every time I build or do I have a wrong setup.

**Comment by ronzulu** [comment]
Personally I just copy the file from the build folder to the appropriate Obsidian plug-in folder. 

Then within Obsidian reload the plug-in. Probably not the most efficient way!

**Reply by 4Source** [reply]
Ok not the answer I hoped for. That is indeed not really effective so I need to ask me why is it setup this way.

**Reply by lightningRalf** [reply]
Best use a 'test' vault for testing plugins in obsidian.
And if you want to go further - you can go into testing with jest. But you have to mock obsidian functionality, and that is not really that nice to set it up ;)
